src/hooks/useVotingContract.ts

```diff
@@ -29,13 +29,14 @@ export default function useVotingContract(
           network.chainId.toString()
         );
         setVotingContract(contract);
-        if (hotAddress && voterAddress) {
-          const dvc = createDesignatedVotingContractInstance(
-            signer,
-            voterAddress
-          );
-          setDesignatedVotingContract(dvc);
-        }
+      }
+      // we want this to be able to change when user switches accounts. previously this was only being able to be set once.
+      if (signer && hotAddress && voterAddress) {
```

Comment:
this was moved out of a nested if statement and into its own block. If you check the previous code you will see that if the outer if statement ran without running the inner statement, the inner statement would not have a chance to ever update again.